templates/types/streaming/fastapi/app/api/routers/chat.py

```diff
@@ -38,14 +39,27 @@ class _SourceNodes(BaseModel):
     metadata: Dict[str, Any]
     score: Optional[float]
     text: str
+    url: Optional[str]
 
     @classmethod
     def from_source_node(cls, source_node: NodeWithScore):
+        metadata = source_node.node.metadata
+        url = metadata.get("URL")
+        
+        if not url:
+            file_name = metadata.get("file_name")
+            url_prefix = os.getenv("FILESERVER_URL_PREFIX")
+            if not url_prefix:
+                print("Warning: FILESERVER_URL_PREFIX not set in environment variables")
```

Comment:
Please use logger instead of print(). 
```suggestion
                logger.warn("Warning: FILESERVER_URL_PREFIX not set in environment variables")
```
And adding the uvicorn logger: `logger = logging.getLogger("uvicorn")`.